name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/columns/DividendPaymentColumn.java

```diff
@@ -0,0 +1,170 @@
+package name.abuchen.portfolio.ui.views.columns;
+
+import java.time.LocalDate;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.eclipse.jface.viewers.ColumnLabelProvider;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+
+import name.abuchen.portfolio.model.Adaptor;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityEvent.DividendEvent;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.ui.Messages;
+import name.abuchen.portfolio.ui.util.Colors;
+import name.abuchen.portfolio.ui.util.viewers.Column;
+import name.abuchen.portfolio.ui.util.viewers.ColumnViewerSorter;
+import name.abuchen.portfolio.ui.util.viewers.DateLabelProvider;
+
+public class DividendPaymentColumn
+{
+
+    static Column createNextDividendExDateColumn()
+    {
+        Column column = new Column("nexdd", Messages.ColumnDividendsNextExDate, SWT.LEFT, 80); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextExDate_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+
+        LocalDate now = LocalDate.now();
+
+        Function<Object, LocalDate> dataProvider = element -> {
+            Security sec = Adaptor.adapt(Security.class, element);
+            if (sec == null)
+            {
+                return null; //
+            }
+            return sec.getEvents().stream() //
+                            .filter(DividendEvent.class::isInstance) //
+                            .map(e -> ((DividendEvent) e).getDate()) //
+                            .sorted(Comparable::compareTo) //
+                            .filter(d -> d.isAfter(now)) //
+                            .findFirst().orElse(null);
+        };
+
+        column.setLabelProvider(new DateLabelProvider(dataProvider)
+        {
+            @Override
+            public Color getBackground(Object element)
+            {
+                LocalDate date = dataProvider.apply(element);
+                if (date == null)
+                {
+                    return null; //
+                }
+                if (!date.equals(LocalDate.now()))
+                {
+                    return null; //
+                }
+                return Colors.theme().greenBackground();
```

Comment:
```suggestion
if (date == null || !date.equals(LocalDate.now())) {
    return null;
}

return Colors.theme().greenBackground();
```

Reply:
We can change that, yes. I prefer these separate checks (easier to follow for people trying to make sense of it and you can see what part isn't covered in coverage-reports) but this particular case is simple enough to be put into one if-statement.